File: console/src/main/java/com/jraska/console/UserTouchingListener.java
package com.jraska.console;

import android.view.MotionEvent;
import android.view.View;

final class UserTouchingListener implements View.OnTouchListener {
  private boolean _userHolds;

  boolean isUserTouching() {
    return _userHolds;
  }

  @Override
  public boolean onTouch(View v, MotionEvent event) {
    switch (event.getActionMasked()) {
      case MotionEvent.ACTION_MOVE:
        _userHolds = true;
        break;

      case MotionEvent.ACTION_UP:
      case MotionEvent.ACTION_CANCEL:
        _userHolds = false;
    }

    return false;
  }
}

File: console/src/main/java/com/jraska/console/UserTouchingListener.kt
package com.jraska.console

import android.view.MotionEvent
import android.view.View

internal class UserTouchingListener : View.OnTouchListener {
  var isUserTouching: Boolean = false
    private set

  override fun onTouch(v: View, event: MotionEvent): Boolean {
    when (event.actionMasked) {
      MotionEvent.ACTION_MOVE -> isUserTouching = true
      MotionEvent.ACTION_UP, MotionEvent.ACTION_CANCEL -> isUserTouching = false
    }

    return false
  }
}
